BitwardenShared/Core/Platform/Services/ClientService.swift

```diff
@@ -199,28 +211,31 @@ class DefaultClientService: ClientService {
     /// Returns a user's client if it exists. If the user has no client, create one and map it to their user ID.
     ///
     ///
-    /// If there is no active user/there are no accounts, return the original client.
+    /// If there is no active user/there are no accounts, return a new client.
     /// This could occur if the app is launched from a fresh install.
     ///
-    /// - Parameter userId: A user ID for which a `Client` is mapped to or will be mapped to.
+    /// - Parameters:
+    ///   - userId: A user ID for which a `Client` is mapped to or will be mapped to.
+    ///   - isPreAuth: Whether the client is being used for a user prior to authentication (when
+    ///     the user's ID doesn't yet exist).
     /// - Returns: A user's client.
     ///
-    private func client(for userId: String?) async throws -> BitwardenSdkClient {
-        do {
-            let userId = try await stateService.getAccountIdOrActiveId(userId: userId)
-
-            // If the user has a client, return it.
-            guard let client = userClientArray[userId] else {
-                // If not, create one, map it to the user, then return it.
-                let newClient = createAndMapClient(for: userId)
-                return newClient
-            }
-            return client
-        } catch StateServiceError.noAccounts, StateServiceError.noActiveAccount {
-            // If there is no active account, or if no accounts exist,
-            // return the original client.
+    private func client(for userId: String?, isPreAuth: Bool = false) async throws -> BitwardenSdkClient {
+        guard !isPreAuth else {
+            // If this client is being used for a new user prior to authentication, a user ID doesn't
+            // exist for the user to map the client to, so return a new client.
             return clientBuilder.buildClient()
         }
+
+        let userId = try await stateService.getAccountIdOrActiveId(userId: userId)
+
+        // If the user has a client, return it.
+        guard let client = userClientArray[userId] else {
+            // If not, create one, map it to the user, then return it.
+            let newClient = createAndMapClient(for: userId)
+            return newClient
+        }
+        return client
```

Comment:
🤔 Shouldn't we maintain the `catch StateServiceError.noAccounts, StateServiceError.noActiveAccount` logic in case this happens? Perhaps because of passing wrongly the `isPreAuth` flag.

Reply:
My thinking on removing it was that it would help us catch any potential uses where `isPreAuth` flag isn't set, but should be. Since if there's an active account and `isPreAuth` isn't set, it will just silently use the account's client.

But it would be safer to keep this, so I'll add it back and maybe see about logging an error.